app/models/manageiq/providers/oracle_cloud/cloud_manager/cloud_database.rb

```diff
@@ -1,2 +1,167 @@
+        },
+        {
+          :component  => 'text-field',
+          :id         => 'shape_name',
+          :name       => 'shape_name',
+          :label      => _('Database Shape Name'),
+          :isRequired => true,
+          :condition  => {
+            :when => 'database',
+            :is   => 'mysql',
+          },
+        },
+        {
+          :component    => 'select',
+          :name         => 'subnet_name',
+          :id           => 'subnet_name',
+          :label        => _('Cloud Subnet'),
+          :includeEmpty => true,
+          :isRequired   => true,
+          :condition    => {
+            :when => 'database',
+            :is   => 'mysql',
+          },
+          :options      => ems.cloud_subnets.map do |cs|
+            {
+              :label => cs[:name],
+              :value => cs[:name]
+            }
+          end
+        },
+        {
+          :component  => 'text-field',
+          :name       => 'cpu_cores',
+          :id         => 'cpu_cores',
+          :label      => _('CPU Cores'),
+          :type       => 'number',
+          :step       => 1,
+          :isRequired => true,
+          :condition  => {
+            :when => 'database',
+            :is   => 'oracle',
+          },
+        },
+        {
+          :component  => 'text-field',
+          :name       => 'storage',
+          :id         => 'storage',
+          :label      => _('Storage Size (in Terabytes)'),
+          :type       => 'number',
+          :step       => 1,
+          :isRequired => true,
+          :condition  => {
+            :when => 'database',
+            :is   => 'oracle',
+          },
+        },
+      ],
+    }
+  end
+
+  def self.raw_create_cloud_database(ext_management_system, options)
+    options[:database] == 'oracle' ? create_oracle_database(ext_management_system, options) : create_mysql_database(ext_management_system, options)
+  rescue => err
+    _log.error("cloud database=[#{options[:name]}], error: #{err}")
+    raise
+  end
+
+  def self.create_oracle_database(ext_management_system, options)
+    require 'oci/database/database'
+
+    ext_management_system.with_provider_connection(:service => 'Database::DatabaseClient') do |connection|
+      connection.create_autonomous_database(
+        OCI::Database::Models::CreateAutonomousDatabaseDetails.new(
+          :db_name                  => options[:name],
+          :display_name             => options[:name],
+          :admin_password           => options[:password],
+          :cpu_core_count           => options[:cpu_cores],
+          :data_storage_size_in_tbs => options[:storage],
+          :compartment_id           => options[:compartment_id]
+        )
+      )
+    end
+  end
+
+  def self.create_mysql_database(ext_management_system, options)
+    require 'oci/mysql/mysql'
+
+    subnet_id = ManageIQ::Providers::OracleCloud::NetworkManager::CloudSubnet.find_by(:name => options[:subnet_name])[:ems_ref]
```

Comment:
Then this can be:
```suggestion
    subnet_id = ManageIQ::Providers::OracleCloud::NetworkManager::CloudSubnet.find(options[:subnet]).ems_ref
```

Reply:
(Main reason being that a database-wide query for subnet by name isn't guaranteed to be unique)

Reply:
Ah didn't think about that 👌